File: Reflow/Provider/IDataLink.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;

namespace Reflow.Provider
{
    public interface IDataLink
    {
        bool Initialize(string connectionString);
        bool IsConnected { get; }
        bool Connect();
        bool Disconnect();
        ColumnDefinition[] GetSchema(string query);
        bool CreateTable(string tableName, ColumnDefinition[] columns, bool shouldDropExisting);
    }
}

File: Java/src/main/java/com/nuarca/etl/provider/IDataLink.java
package com.nuarca.etl.provider;

import com.nuarca.etl.ColumnDefinition;

public interface IDataLink {

    boolean initialize(String connectionString) throws Exception;

    boolean getIsConnected() throws Exception;

    boolean connect() throws Exception;

    boolean disconnect() throws Exception;

    ColumnDefinition[] getSchema(String query) throws Exception;

    boolean createTable(String tableName, ColumnDefinition[] columns, boolean shouldDropExisting) throws Exception;
}


